### PR TITLE
Remove deprecated player & auth files backends

### PR DESCRIPTION
**Pull request by appgurueu**
Removes the long deprecated player & auth files backends.

*Oh no! Does this mean I'll lose (access to) my data?*
No, even in the extremely unlikely event that you have worlds using the "files" backend for auth or player data, you could still use an older Minetest version to migrate to a newer backend, such as sqlite3.

## How to test

:eye: :lips: :eye:

### CODE REVIEW

File: src/database/database-files.cpp [hunk 1]
```diff
@@ -27,355 +27,6 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "util/string.h"
 #include <cassert>
```
Comment (JosiahWI) [inline src/database/database-files.cpp:28]: Headers `"remoteplayer.h"`, `"settings.h"`, `"server/player_sao.h"`, and `<cassert>` are no longer needed.

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Please update the documentation in `doc/lua_api.md` line 5252 and `doc/world_format.txt` line 132.

**Comment by appgurueu** [comment]
Thanks for the review! `doc/lua_api.md` refers to the mod storage files backend as far as I can see.

**Comment by sfan5** [comment]
Personally I have just been ignoring these warnings since they've been added, so here's an approach worth considering:
* Turn the warning into an error (refuse to start up)
* Keep the backend code as read-only

This will allow users to easily migrate their stuff without having to figure out where to obtain an older Minetest version.

**Comment by lhofhansl** [comment]
> Personally I have just been ignoring these warnings since they've been added

Same here. I agree with @sfan5 to turn these into errors first.

**Comment by Zughy** [comment]
@appgurueu 👀

**Comment by appgurueu** [comment]
This PR has to rest until I do the thing sfan suggested (I've put it on my personal to-do), we release it, and a while later we can remove this. I will reopen it when / if that time comes.

**Comment by sfan5** [comment]
Another thought that just crossed my mind is: This affects singleplayer worlds too, and if you consider platforms like Android then it's **impossible** for some people to open the shell and run `minetest --server --migrate-players sqlite3 --worldname abc` (never mind the catastrophic UX).
So we do need some kind of automatic migration, perhaps restricted to single player only.

**Comment by Desour** [comment]
Or add buttons in main menu to migrate worlds.

**Comment by Desour** [comment]
> > Or add buttons in main menu to migrate worlds.
> 
> Please don't, that would be very bad UX. Players shouldn't have to bother about backends and data migration.

But we also bother them with warnings.

If done properly, this shouldn't really confuse players.
Just add some yellow text saying "23 of your worlds use deprecated database backends, that will be dropped in the future.", and a button "Migrate them" below, which then migrates all your worlds.
While this is not automatic, it's still semi-automatic. And it will not lead to weird significant longer world join times or other issues.
(I'd also like additional per-world info and buttons, for advanced users.)

**Pull request closed without merging**